src/jobs/consumers/notifications/smtp-notifications.job.consumer.ts

```diff
@@ -4,24 +4,36 @@ import { InjectRepository } from '@nestjs/typeorm';
 import { Repository } from 'typeorm';
 import { Notification } from 'src/models/notifications/entities/notification.entity';
 import { DeliveryStatus } from 'src/common/constants/notifications';
+import { SmtpService } from 'src/services/email/smtp/smtp.service';
+import * as nodemailer from 'nodemailer';
 
 @Processor('smtpNotifications')
 export class SmtpNotificationConsumer {
   constructor(
     @InjectRepository(Notification)
     private readonly notificationRepository: Repository<Notification>,
+    private readonly smtpService: SmtpService,
   ) {}
 
   @Process()
   async processSmtpNotificationQueue(job: Job): Promise<void> {
-    const notification = job.data;
+    const id = job.data;
+    const notification = (await this.getNotificationById(id))[0];
     try {
-      // TODO: Send the email, update notification values
+      this.smtpService.sendEmail(notification.data as nodemailer.SendMailOptions);
       notification.deliveryStatus = DeliveryStatus.SUCCESS;
     } catch (error) {
       notification.deliveryStatus = DeliveryStatus.FAILED;
     } finally {
       await this.notificationRepository.save(notification);
     }
   }
+
+  getNotificationById(id: number): Promise<Notification[]> {
```

Comment:
@vikas-kyatannawar where should methods like this go?

We used to keep such methods in express MVC framework in model files. Since they are reusable.

Where should they go now?

Reply:
This should be present in notification.service.ts

Reply:
This has been done in the latest commit